src/grammar/grammar.ts

```diff
@@ -797,8 +851,46 @@ export class StateStack implements StackElementDef {
 		}
 		return false;
 	}
+
+	public toStateStackFrame(): StateStackFrame {
+		return {
+			ruleId: ruleIdToNumber(this.ruleId),
+			enterPos: this._enterPos,
+			anchorPos: this._anchorPos,
```

Comment:
It isn't necessary to serialize `enterPos` and `anchorPos`, they are used as "storage" during tokenization of a line, but they don't hold any inherent meaning across lines.